slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/OldPlainSQLTest.scala

```diff
@@ -110,15 +110,55 @@ class OldPlainSQLTest extends TestkitTest[JdbcTestDB] {
     println("User for ID 2: "+res)
     assertEquals(User(2,"guest"), res)
 
-    val s1 = sql"select id from USERS where name = ${"szeiger"}".as[Int]
-    val s2 = sql"select id from USERS where name = '#${"guest"}'".as[Int]
-    assertEquals("select id from USERS where name = ?", s1.getStatement)
+    val s1 = sql"select id from USERS where name = '${"szeiger"}'".as[Int]
+    val s2 = sql"select id from USERS where name = '${"guest"}'".as[Int]
+    assertEquals("select id from USERS where name = 'szeiger'", s1.getStatement)
```

Comment:
Something like this should work: Change SQLActionBuilder to build build the query string dynamically:

```
case class SQLActionBuilder(sqlParts: Seq[Any], unitPConv: SetParameter[Unit]) {
  def as[R](implicit rconv: GetResult[R]): SqlStreamingAction[Effect, Vector[R], R] = {
    val sql = sqlParts.iterator.map(String.valueOf).mkString
    new StreamingInvokerAction[Effect, Vector[R], R] {
      def statements = List(sql)
      protected[this] def createInvoker(statements: Iterable[String]) = new StaticQueryInvoker[Unit, R](statements.head, unitPConv, (), rconv)
      protected[this] def createBuilder = Vector.newBuilder[R]
    }
  }
  def asUpdate = as[Int](GetResult.GetUpdateValue)
}
```

The individual parts have to be produced as Exprs to be spliced in. Anything after a `#` goes into the query parts, other arguments go into pconv.